framework/Samplers/MCMC/MCMC.py

```diff
@@ -0,0 +1,293 @@
+    self._updateValues = {} # dict stores input variables values for the current MCMC iteration, i.e. {var:val}
+    self._proposal = {} # dict stores the proposal distributions for input variables, i.e. {var:dist}
+    self._tune = 0      # integers indicate how many samples will be discarded
+    self._likelihood = None # stores the output from the likelihood
+    self._availProposal = {'normal': Distributions.Normal(0.0, 1.0),
+                           'uniform': Distributions.Uniform(-1.0, 1.0)} # available proposal distributions
+    self._acceptDist = Distributions.Uniform(0.0, 1.0) # uniform distribution for accept/rejection purpose
+    # assembler objects
+    self.addAssemblerObject('proposal', '-n', True)
+
+  def localInputAndChecks(self, xmlNode, paramInput):
+    """
+      unfortunately-named method that serves as a pass-through for input reading.
+      comes from inheriting from Sampler and _readMoreXML chain.
+      @ In, xmlNode, xml.etree.ElementTree.Element, xml element node (don't use!)
+      @ In, paramInput, InputData.ParameterInput, parameter specs interpreted
+      @ Out, None
+    """
+    # this is just a passthrough until sampler gets reworked or renamed
+    self.handleInput(paramInput)
+
+  def handleInput(self, paramInput):
+    """
+      Read input specs
+      @ In, paramInput, InputData.ParameterInput, parameter specs interpreted
+      @ Out, None
+    """
+    likelihood = paramInput.findFirst('likelihood')
+    if likelihood is not None:
+      self._likelihood = likelihood.value
+    else:
+      self.raiseAnError(IOError, "likelihood is required, but not provided!")
+    init = paramInput.findFirst('samplerInit')
+    if init is not None:
+      # limit
+      limit = init.findFirst('limit')
+      if limit is not None:
+        self.limit = limit.value
+      else:
+        self.raiseAnError(IOError, 'MCMC', self.name, 'needs the limit block (number of samples) in the samplerInit block')
+      # initialSeed
+      seed = init.findFirst('initialSeed')
+      if seed is not None:
+        self.initSeed = seed.value
+      else:
+        self.initSeed = randomUtils.randomIntegers(0,2**31,self)
+      tune = init.findFirst('tune')
+      if tune is not None:
+        self._tune = tune.value
+    else:
+      self.raiseAnError(IOError, 'MCMC', self.name, 'needs the samplerInit block')
+    # variables additional reading
+    for varNode in paramInput.findAll('variable'):
+      var = varNode.parameterValues['name']
+      initNode = varNode.findFirst('initial')
+      if initNode:
+        self._initialValues[var] = initNode.value
+      else:
+        self._initialValues[var] = None
+      proposal = varNode.findFirst('proposal')
+      if proposal:
+        self._proposal[var] = proposal.value
+      else:
+        self._proposal[var] = None
+    # TargetEvaluation Node (Required)
+    targetEval = paramInput.findFirst('TargetEvaluation')
```

Comment:
I believe the AdaptiveSampler does this for you, so it should not be needed here if you call the base class, right?